Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -16,6 +16,118 @@ import SwiftParser
 import SwiftSyntax
 
 class Frontend {
+  /// Provides formatter configurations for given `.swift` source files, configuration files or configuration strings.
+  struct ConfigurationProvider {
+    /// Loads formatter configuration files and chaches them in memory.
+    private var configurationLoader: ConfigurationLoader = ConfigurationLoader()
```

Comment:
Actually, I don’t think this needs to be a member at all. A local variable in `provide` should be sufficient, right?

Reply:
It stores the cache of previously loaded configurations. So it wouldn't work as a local variable (as in: stack-allocated inside `provide`, if that is what you meant?).